rpmio/rpmkeyring.cc

```diff
@@ -363,52 +372,59 @@ pgpDigParams rpmPubkeyPgpDigParams(rpmPubkey key)
     return params;
 }
 
-static rpmPubkey findbySig(rpmKeyring keyring, pgpDigParams sig)
+rpmRC rpmKeyringVerifySig2(rpmKeyring keyring, pgpDigParams sig, DIGEST_CTX ctx, rpmPubkey * keyptr)
 {
+    rpmRC rc = RPMRC_FAIL;
     rpmPubkey key = NULL;
+    std::vector<std::pair<int, std::string>> results = {};
```

Comment:
And here too - the results vector is only needed inside the main if, so that's where it belongs. And key inside the only code that can actually end with it being set.

Reply:
Variables being in the most local scope is something that really belongs to self-review part, FWIW.